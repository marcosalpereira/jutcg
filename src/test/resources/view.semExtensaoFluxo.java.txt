package br.view.correcao;

public class ManagedBean {

    private List<Person> persons;

    @In
    private Facade facade;

    public void pesquisar() {
        persons = facade.searchPersons();
    }

}
