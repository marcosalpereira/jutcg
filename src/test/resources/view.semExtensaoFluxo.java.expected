package br.view.correcao;

import static org.easymock.EasyMock.*;
import org.junit.*;
import java.util.*;
import static org.junit.Assert.*;

/**
 * Testes unitarios para a classe {@link ManagedBean}.
 */
public class TestManagedBean extends br.view.JsfTestCase {

/**
 * Classe sendo testada {@link ManagedBean}.
 */
private ManagedBean managedBean;

@Override
public void setUp() throws Exception {
  super.setUp();
  managedBean = new ManagedBean();
}

/**
 * Cria o mock {@link Facade} e seta na classe sendo testada.
 * @return o mock criado
 */
private Facade criarMockFacade() {
  Facade mock = createStrictMock(Facade.class);
  managedBean.setFacade(mock);
  return mock;
}

/**
 * Teste para o metodo {@link ManagedBean#pesquisar()}.
 */
@Test
public void testPesquisar() {

// Configurando mock
Facade facade = criarMockFacade();
expect(facade.searchPersons())
  .andReturn(persons);
replay(facade);

// invocar metodo sendo testado
managedBean.pesquisar();

// checar estados dos mocks
verify(facade);

// verificacoes do resultado do metodo sendo testado
final List<Person> getPersonsExpected;
assertEquals(getPersonsExpected, managedBean.getPersons());
}
}
